### ISSUE TITLE
Use a single table for user mapping

### ISSUE THREAD

**Issue by allardy** [botpress/messaging#40]
There is `web_user_map` to map channel web's visitor ID to Messaging UID, and you use `msg_kvs` to store channel-teams user ID to Messaging UID.

Channels each have their unique ID to identify a visitor.  Each entry in `msg_users` represent an association `userID` - `botId`, why don't you add the channel's internal ID to this table instead ? 

We could also add a "metadata" field which could be used to store additional data if required.

I see absolutely no reason to have a table "msg_kvs" in this service.

**Comment by samuelmasse** [comment]
web_user_map is from the main botpress repo. Messaging tables start with msg_

**Comment by samuelmasse** [comment]
The kvs in teams is used to stored some data that we need to answer messages. The teams package uses an object oriented approach, so just saving the id to the conversation doesn't work here. There may be a possibility to fetch that instead of storing it in the kvs, but for now we kept it like it was already.

**Comment by allardy** [comment]
But what is the purpose of `visitorId` now that we have a real unique User ID ? I really think that users on channel-web should directly use the new unique user ID instead of having a mapping. 

For existing users, we could query channel-web with its `visitorId` to obtain its messaging UUID, then continue the conversation using the UUID.

**Comment by samuelmasse** [comment]
![diagram](https://user-images.githubusercontent.com/59894025/126666997-e0bac355-3452-4a3c-9f14-fe8bb92cd310.png)

Here is a diagram of the db. I generated it using pgAdmin. It's a bit hard to see the relations since they overlap on each other so I'll make a better one.

I organized the graph in 5 group of tables. The first group at the left is msg_clients and msg_webhooks. A client represents an api user. For example a bot has always one client (with clientId and clientToken). The client has webhooks that are stored in the msg_webhooks table.

The next section is the tables that store our users, conversations and messages. User in this case refers to a chat user. A message has an author (user) and a conversation. A conversation is owned by a user.

The next section is the mapping section. This section maps users and conversations from the previous section to senders and threads from the next section. Mapping only occurs for external channels (so for example channel-web doesn't have any mapping).

The next section is the foreign data section. It contains data that was obtained from foreign services (such as facebook, telegram, twilio, etc). Right now it only contains the id of each item, although it could also store all sorts of metadata in the future. You have the msg_tunnel that simply represents a conjunction of a client and a channel. Then you have msg_identities which hold the "name" of the endpoint you contacted on a channel (for example on twilio this would be the phone number used by twilio). Then you have msg_senders which hold information on the "writer" of the message (this is always defined to a value on all channels). Finally you have msg_threads which holds informations on which conversation you are on with the bot (in the case where the same user can hold more than one conversation with a bot).

All of the previous sections mentioned is data that is considered purely client owned. Meaning that if a client decides to stop using messaging, we can trace back all of it's data on the server and delete it. A provider could also end up being client owned as well if a client points to it (because of sandboxes, providers aren't client owned 100% of the time).

The last section is the provider section. It simply provides a connection to external services. You have a provider which can hold up to 1 connection on each channel. A connection to a channel is called a conduit. A bot will usually have one provider and one client. The distinction between providers and clients exists because later on we'll set up sandboxes to make trying channels easier, and these sandboxes will be providers only, that dynamically connect to multiple clients (this is already 80% done).

The msg_channels table is a convenient table to have foreign keys on our available channels. The channel ids are constant and defined in their channel.ts file. A channel also has a friendly name that is defined there as well.
